src/main/java/io/wisetime/connector/fetch_client/FetchClient.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2019 Practice Insight Pty Ltd. All Rights Reserved.
+ */
+
+package io.wisetime.connector.fetch_client;
+
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.wisetime.connector.TimePosterRunner;
+import io.wisetime.connector.api_client.ApiClient;
+import io.wisetime.connector.api_client.PostResult;
+import io.wisetime.connector.integrate.WiseTimeConnector;
+import io.wisetime.generated.connect.TimeGroupStatus;
+
+/**
+ * Implements a fetch based approach to retrieve time groups.
+ *
+ * @author pascal.filippi@staff.wisetime.com
+ */
+public class FetchClient implements Runnable, TimePosterRunner {
+
+  private static final int MAX_MINS_SINCE_SUCCESS = 10;
+
+  private static final Logger log = LoggerFactory.getLogger(FetchClient.class);
+
+  private ApiClient apiClient;
+
+  private TimeGroupIdStore timeGroupIdStore;
+
+  private String fetchClientId;
+
+  private int limit;
+
+  private WiseTimeConnector connector;
+
+  private ExecutorService postTimeExecutor;
+
+  private final AtomicReference<DateTime> lastSuccessfulRun;
+
+  private ExecutorService fetchClientExecutor;
+
+  public FetchClient(ApiClient apiClient,
+                     WiseTimeConnector connector,
+                     TimeGroupIdStore timeGroupIdStore,
+                     String fetchClientId,
+                     int limit) {
+    this.apiClient = apiClient;
+    this.connector = connector;
+    this.timeGroupIdStore = timeGroupIdStore;
+    this.fetchClientId = fetchClientId;
+    this.limit = limit;
+    this.postTimeExecutor = Executors.newFixedThreadPool(limit);
+    this.lastSuccessfulRun = new AtomicReference<>(DateTime.now());
+  }
+
+  @Override
+  public void run() {
+    while (!Thread.currentThread().isInterrupted()) {
+      try {
+        apiClient.fetchTimeGroups(fetchClientId, limit).forEach(timeGroup ->
+            postTimeExecutor.submit(() -> {
+              if (!timeGroupIdStore.alreadySeen(timeGroup.getGroupId())) {
```

Comment:
Is there any scenario where a `TimeGroup` is actually left "seen" in the DB? It looks like it gets deleted in every case - success, failure, exception.